src/Analysis/Ast/Impl/Analyzer/PythonAnalyzer.cs

```diff
@@ -144,24 +148,36 @@ public Task WaitForCompleteAnalysisAsync(CancellationToken cancellationToken = d
             }
         }
 
-        public void EnqueueDocumentForAnalysis(IPythonModule module, int bufferVersion) {
-            var key = new AnalysisModuleKey(module);
+        public void EnqueueDocumentForAnalysis(IPythonModule module, PythonAst ast, int bufferVersion) {
             PythonAnalyzerEntry entry;
+            AnalysisModuleKey key;
             int version;
+
             lock (_syncObj) {
+                key = new AnalysisModuleKey(module);
                 version = _version + 1;
                 if (_analysisEntries.TryGetValue(key, out entry)) {
                     if (entry.BufferVersion >= bufferVersion) {
                         return;
                     }
+
+                    // It is possible that parsing request for the library has been started when document is open,
```

Comment:
This happens when `Go to definition` is done by `Ctrl+Click`. We got three events in a row: `didOpen`, `didClose`, `didOpen`.